internal/wasi/wasi.go

```diff
@@ -306,10 +336,125 @@ type SnapshotPreview1 interface {
 	// TODO: wasi.FdFilestatSetSize
 	// TODO: wasi.FdFilestatSetTimes
 	// TODO: wasi.FdPread
-	// TODO: wasi.FdPrestatGet
-	// TODO: wasi.FdPrestatDirName
+
+	// FdPrestatGet is the WASI function to return the prestat data of a file descriptor.
+	// This returns ErrnoBadf if the fd is invalid.
+	//
+	// * fd - the file decriptor to get the prestat
+	// * resultPrestat - the offset to write the result Prestat struct
+	//
+	// TODO: Define Prestat and PrestatDir, and update the example below.
+	// For example, if fd 3 is a file with a prestat
+	//   /* Note: Prestat is not defined in wazero yet. This is an example for explanation. */
+	//   Prestat {
+	//     Tag: PrestatDir /* 0 (type: uint8, offset: 0) */
+	//     PrNameLen: 3    /* 3 (type: uint32, offset: 4) */
+	//   }
+	// and FdPrestatGet parameters fd = 3, resultPrestat = 1, we expect `ctx.Memory` to contain:
+	//
+	//                     padding   uint32le
+	//          uint8 --+  +-----+  +--------+
```

Comment:
good example. Instead of defining a datatype, you can have other docs refer here to the bit layout of the tag and prNameLen. Reason is that we cannot pass structs and don't want to repeat ourselves, so I think this is the better of evils for now.